python/pyarrow/_dataset.pyx

```diff
@@ -1311,38 +1284,74 @@ cdef class ParquetFileWriteOptions(FileWriteOptions):
         self._set_arrow_properties()
 
 
+cdef set _PARQUET_READ_OPTIONS = {'dictionary_columns'}
+
+
 cdef class ParquetFileFormat(FileFormat):
 
     cdef:
         CParquetFileFormat* parquet_format
 
-    def __init__(self, read_options=None):
+    def __init__(self, read_options=None,
+                 default_fragment_scan_options=None, **kwargs):
```

Comment:
This is mostly, but not fully, backwards compatible (something like `ds.ReadOptions(use_buffered_stream=True)` no longer works) - is that an issue?